include/dlaf/eigensolver/tridiag_solver/merge.h

```diff
@@ -681,22 +733,28 @@ void mergeSubproblems(const SizeType i_begin, const SizeType i_split, const Size
   auto rots =
       applyDeflation(i_begin, i_end, scaled_rho, std::move(tol), ws_hm.i2, ws_h.d0, ws_hm.z0, ws_h.c);
 
-  // ---
-
   applyGivensRotationsToMatrixColumns(i_begin, i_end, std::move(rots), ws.e0);
-  // Placeholder for rearranging the eigenvectors: (local permutation)
-  copy(idx_loc_begin, sz_loc_tiles, ws.e0, ws.e1);
+
+  initSortByColTypeIndex(i_begin, i_end, ws_h.c, ws_hm.i5);
+  copy(idx_begin_tiles_vec, sz_tiles_vec, ws_hm.i5, ws.i5);
+  dlaf::permutations::permute<B, D, T, Coord::Col>(i_begin, i_end, ws.i5, ws.e0, ws.e1);
 
   // Step #2
   //
-  //    i2 (in)  : initial <--- pre_sorted
-  //    i3 (out) : initial <--- deflated
+  //    i2 (in)  : initial  <--- pre_sorted
+  //    i5 (in)  : initial  <--- sorted by coltype
+  //    i3 (out) : initial  <--- deflated
+  //    i4 (out) : deflated <--- sorted by coltype
   //
   // - reorder `d0 -> d1`, `z0 -> z1`, using `i3` such that deflated entries are at the bottom.
-  // - solve the rank-1 problem and save eigenvalues in `d0` and `d1` (copy) and eigenvectors in `e2`.
+  // - compute permutation `i4`: sorted by col type ---> deflated
+  // - solve rank-1 problem and save eigenvalues in `d0` and `d1` (copy) and eigenvectors in `e2` (sorted
+  // by coltype)
   // - set deflated diagonal entries of `U` to 1 (temporary solution until optimized GEMM is implemented)
   //
-  auto k = stablePartitionIndexForDeflation(i_begin, i_end, ws_h.c, ws_hm.i2, ws_h.i3) | ex::split();
+  auto k =
+      ex::ensure_started(stablePartitionIndexForDeflation(i_begin, i_end, ws_h.c, ws_hm.i2, ws_h.i3)) |
+      ex::split();
```

Comment:
@msimberg do you think it is better, in general, to `ensure_started`/`start_detached` tasks? because this (but there are also others) for starting relied on it being a dependency for next tasks.

Reply:
Generally speaking, avoid `ensure_started` unless it's required for progress since it causes an extra heap allocation. However, as we've found out we can get into situations where eagerly starting work is required, and in those cases you of course have to use `ensure_started`.